File: packages/shared/src/roles.ts
/**
 * Sistem genelinde kullanılan roller.
 * Backend JWT içindeki `role` claim'i ile birebir eşleşir.
 */
export const ROLES = {
  SUPER_ADMIN: 'super_admin',
  COMPANY_ADMIN: 'company_admin',
  OPERATIONS_MANAGER: 'operations_manager',
  DRIVER: 'driver',
  PASSENGER: 'passenger',
} as const;

export type Role = (typeof ROLES)[keyof typeof ROLES];

export const ALL_ROLES: Role[] = Object.values(ROLES);

/**
 * Her rolün giriş sonrası yönlendirileceği Expo Router route group'u.
 * operations_manager, company_admin ile aynı yönetici arayüzünü paylaşır.
 */
export const ROLE_HOME_SEGMENT: Record<Role, string> = {
  [ROLES.SUPER_ADMIN]: '(super-admin)',
  [ROLES.COMPANY_ADMIN]: '(company-admin)',
  [ROLES.OPERATIONS_MANAGER]: '(company-admin)',
  [ROLES.DRIVER]: '(driver)',
  [ROLES.PASSENGER]: '(passenger)',
};

/** İnsan tarafından okunabilir rol etiketleri (Türkçe UI). */
export const ROLE_LABELS: Record<Role, string> = {
  [ROLES.SUPER_ADMIN]: 'Süper Admin',
  [ROLES.COMPANY_ADMIN]: 'Yönetici',
  [ROLES.OPERATIONS_MANAGER]: 'Operasyon Yetkilisi',
  [ROLES.DRIVER]: 'Şoför',
  [ROLES.PASSENGER]: 'Yolcu',
};

export function isRole(value: unknown): value is Role {
  return typeof value === 'string' && (ALL_ROLES as string[]).includes(value);
}

File: apps/mobile/lib/core/constants/roles.dart
/// Sistem rolleri — backend JWT `role` claim'i ile birebir eşleşir.
enum Role {
  superAdmin('super_admin'),
  companyAdmin('company_admin'),
  operationsManager('operations_manager'),
  driver('driver'),
  passenger('passenger');

  const Role(this.value);

  /// Backend/DB'de saklanan string değeri.
  final String value;

  static Role fromValue(String value) {
    return Role.values.firstWhere(
      (r) => r.value == value,
      orElse: () => Role.passenger,
    );
  }

  /// Türkçe etiket (UI).
  String get label => switch (this) {
        Role.superAdmin => 'Süper Admin',
        Role.companyAdmin => 'Yönetici',
        Role.operationsManager => 'Operasyon Yetkilisi',
        Role.driver => 'Şoför',
        Role.passenger => 'Yolcu',
      };

  /// Giriş sonrası yönlendirilecek ana rota. operations_manager, yönetici
  /// arayüzünü paylaşır.
  String get homeRoute => switch (this) {
        Role.superAdmin => '/super-admin',
        Role.companyAdmin => '/admin',
        Role.operationsManager => '/admin',
        Role.driver => '/driver',
        Role.passenger => '/passenger',
      };
}
